### ISSUE TITLE
No import suggestions for multi-root workspace with editable installs

### ISSUE THREAD

**Issue by claudius-kienle** [microsoft/pylance-release#3732]
I am using the vs code multi-root workspace feature to open multiple python packages as individual projects in one vscode window. All packages are installed in one virtual environment (pip) as editable install (in `compat mode`).
Some of these packages depend on other packages in the multi-root workspace.

I now face the issue that if I try to import a function in `package-2` that is defined in `package-1`, VS code does not list the import in the `code actions` (`ctrl+.`).  
Currently, I manually have to add the import statement to the top of the file. Additionally, once I've added the import statement, it is detected by pylance ('not defined' error fanishes), so it looks like pylance has some knowledge about the existence of this function.

I searched for existing issues but couldn't find any. Also opening the file, where the function I'm trying to import is defined, didn't help.

I created a minimal example ([https://github.com/claudius-kienle/pylance-import](https://github.com/claudius-kienle/pylance-import)) that can be used to reproduce the behaviour I described above.

I would love some support how to configure pylance that it will list the import statement in the code-actions to import stuff from different workspaces automatically.

Note: I'm only trying to do a standard import, no auto-import (since this is only possible for open files/ current workspace)

## Update

I think I now know the route cause of the problem. Since I installed the packages, they are treated as third party packages and not as user packages. As mentioned in #3281 and #2991, only the top level public symbols are indexed in third party packages. 
When I define `python.analysis.packageIndexDepths: [["", 20, true]]` in the settings, the import shows up in the code actions.

## Suggestion

I think it would help to add a note in the editable installs section in the `troubleshooting.md`, since editable installs are used more often nowadays. Also I am still struggeling to setup the `python.analysis.packageIndexDephts` setting. 
Setting it to `[["", 20, true]]` worked, but I would like to only change the index depth for the packages I installed as editable installs.
I tried to set it to 
```json
"python.analysis.packageIndexDepths": {
            {
                "name": "package-1",
                "depth": 20,
                "includeAllSymbols": true
            }
}
```
but this did not work. Do I have to pass anything else as `name`, because it is an editable install?

**Comment by heejaechang** [comment]
thank you for reporting. let me take a look.

**Comment by heejaechang** [comment]
@claudius-kienle sorry for late reply. 

the reason this didn't work

```
"python.analysis.packageIndexDepths": {
            {
                "name": "package-1",
                "depth": 20,
                "includeAllSymbols": true
            }
}
```

is because you used wrong name. the package name should be `package_1` not `package-1`. once you fix that, it should work. 

but that being said, let me see whether I can improve overall experience better such as modifying reflected faster than what it has now.

**Comment by claudius-kienle** [comment]
Ah yeah you're right, thank you. But still, as you also mentioned, pylance does not update if i update e.g. a function name in package-1 and try to import it in package-2, since it assumes it as being a 3rd party dependency which won't change.

**Comment by heejaechang** [comment]
now, update time is reduced from 10 min to 10 seconds after a file is saved. still, not supporting live update since that can impact typing perf. but still open to feedback once it is released in next version.

**Comment by heejaechang** [comment]
This issue has been fixed in prerelease version 2023.1.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023121-11-January-2023-PreRelease)
